aspnetcore/release-notes/aspnetcore-6.0.md

```diff
@@ -0,0 +1,710 @@
+HTTP logging provides logs of:
+
+* HTTP Request information
+* Common properties
+* Headers
+* Body
+* HTTP Response information
+
+To configure the HTTP logging middleware, specify <xref:Microsoft.AspNetCore.HttpLogging.HttpLoggingOptions>:
+
+[!code-csharp[](aspnetcore-6.0/samples/WebApp1/Program.cs?name=snippet_httplg2&highlight=1,4-13)]
+
+### IConnectionSocketFeature
+
+The <xref:Microsoft.AspNetCore.Connections.Features.IConnectionSocketFeature> request feature provides access to the underlying accept socket associated with the current request. It can be accessed via the <xref:Microsoft.AspNetCore.Http.Features.FeatureCollection> on `HttpContext`.
+
+For example, the following app sets the <xref:System.Net.Sockets.Socket.LingerState> property on the accepted socket:
+
+[!code-csharp[](aspnetcore-6.0/samples/WebApp1/Program.cs?name=snippet_icsf)]
+
+### .NET Hot Reload
+
+Hot Reload minimizes the number of app restarts after code changes. For more information, see [.NET Hot Reload support for ASP.NET Core](https://docs.microsoft.com/aspnet/core/test/hot-reload)
+<!--TODO change above link to <xref:test/hot-reload> when https://github.com/dotnet/AspNetCore.Docs/pull/23659 merges-->
+
+### Generic type constraints in Razor
+
+When defining generic type parameters in Razor using the `@typeparam` directive, generic type constraints can now be specified using the standard C# syntax:
+
+
+### Smaller SignalR, Blazor Server, and MessagePack scripts
+
+ The SignalR, MessagePack, and Blazor Server scripts are now significantly smaller, enabling smaller downloads, less JavaScript parsing and compiling by the browser, and faster start-up. The size reductions:
+
+* `signalr.min.js`: 70%
+* `blazor.server.js`: 45%
+
+The smaller scripts are a result of a community contribution from [Ben Adams](https://twitter.com/ben_a_adams). For more information on the details of the size reduction, see [Ben's GitHub pull request](https://github.com/dotnet/aspnetcore/pull/30320).
+
+### Enable Redis profiling sessions
+
+A community contribution from [Gabriel Lucaci](https://twitter.com/_glucaci) enables Redis profiling session with [Microsoft.Extensions.Caching.StackExchangeRedis](https://www.nuget.org/packages/Microsoft.Extensions.Caching.StackExchangeRedis):
+
+[!code-csharp[](aspnetcore-6.0/samples/WebApp1/Program.cs?name=snippet_red)]
+
+For more information, see [StackExchange.Redis Profiling](https://stackexchange.github.io/StackExchange.Redis/Profiling_v2.html).
+
+### Shadow-copying in IIS
+
+A new feature has been added to the <xref:host-and-deploy/aspnet-core-module> to add support for shadow-copying application assemblies. Currently .NET locks application binaries when running on Windows making it impossible to replace binaries when the app is running. While our recommendation remains to use an [app offline file](xref:host-and-deploy/iis/app-offline), we recognize there are certain scenarios (for example FTP deployments) where it isn’t possible to do so.
```

Comment:
Can we make it explicit that this feature is experimental, and we're looking for feedback?